validator/accounts/status.go

```diff
@@ -0,0 +1,68 @@
+package accounts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/keystore"
+	"go.opencensus.io/trace"
+)
+
+// FetchAccountStatuses fetches validator statuses from the BeaconNodeValidatorClient
+// for each validator public key.
+func FetchAccountStatuses(
+	ctx context.Context,
+	beaconNodeRPCProvider ethpb.BeaconNodeValidatorClient,
+	keyPairs map[string]*keystore.Key,
+) ([]*ethpb.ValidatorStatusResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "validator.FetchAccountStatuses")
+	defer span.End()
+
+	var err error
+	const RequestLimit = 3
+	statuses := make([]*ethpb.ValidatorStatusResponse, 0, len(keyPairs))
+	errorChannel := make(chan error, RequestLimit)
+	statusChannel := make(chan *ethpb.ValidatorStatusResponse, RequestLimit)
+
+	for _, key := range keyPairs {
+		go fetchValidatorStatus(
+			ctx, beaconNodeRPCProvider, key.PublicKey.Marshal(), statusChannel, errorChannel)
+	}
+
+	for i := 0; i < len(keyPairs); i++ {
+		select {
+		case status := <-statusChannel:
+			statuses = append(statuses, status)
+		case e := <-errorChannel:
+			err = e
+		}
+	}
+
+	return statuses, err
+}
+
+func fetchValidatorStatus(
+	ctx context.Context,
+	rpcProvder ethpb.BeaconNodeValidatorClient,
+	pubKey []byte,
+	statusChannel chan *ethpb.ValidatorStatusResponse,
+	errorChannel chan error) {
+	if ctx.Err() == context.Canceled {
+		errorChannel <- errors.Wrap(ctx.Err(), "context has been canceled.")
+		return
+	}
+
+	req := &ethpb.ValidatorStatusRequest{
+		PublicKey: pubKey,
```

Comment:
So if we have N keys then it's N requests to the beacon node. Can we batch this to just 1 request?

Reply:
Yes. Right now it's like this so I can avoid compile errors. 

I'm waiting for prysmaticlabs/ethereumapis#148 to get merged. Once the API for ValidatorStatus is updated, we'll batch public keys into slices of at most `C` keys, so we'll make N/C requests.

And after receiving the results, we'll run a k-way merge sort and then print the results.

Does this sound good?